encoder/vaapiencoder_h264.cpp

```diff
@@ -930,6 +930,15 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time
     bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= keyFramePeriod() || forceKeyFrame);
 
     if (isIdr) {
+        // If the last frame before IDR is B frame, set it to P frame.
```

Comment:
It is discarded by yami(because VAAPI_ENC_REORD_DUMP_FRAMES is not set), not driver.
Maybe I need add flush logic for this case to avoid B frame discarded. But this patch LGTM since it can fix driver crash issue.